### PR TITLE
Make it more clear that you may need to adjust memlock limits

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We had multiple reports from people that tried to use io_uring but did not notice that it failed due memlock settings. Let's make it more clear how to solve the problem

Modifications:

Add section about memlock

Result:

Hopefully its easier for people to understand why using of io_uring failed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,3 +25,46 @@ As an example this is how you would include the dependency in maven:
...
+	at io.netty.incubator.channel.uring.IOUringEventLoopGroup.<init>(IOUringEventLoopGroup.java:86)
+	at io.netty.incubator.channel.uring.IOUringEventLoopGroup.<init>(IOUringEventLoopGroup.java:75)
+	at io.netty.incubator.channel.uring.IOUringEventLoopGroup.<init>(IOUringEventLoopGroup.java:58)
+	at io.netty.incubator.channel.uring.IOUringEventLoopGroup.<init>(IOUringEventLoopGroup.java:44)
+	at ...............NettyServerStart.main(NettyServerStart.java:27)
+	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
+	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
+	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
+	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
+	at com.intellij.rt.execution.application.AppMainV2.main(AppMainV2.java:128)
```
Comment (chrisvest) [inline README.md:55]: I think the stack trace example can be simplified by skipping or compressing some of these frames.

File: README.md [hunk 2]
```diff
@@ -25,3 +25,46 @@ As an example this is how you would include the dependency in maven:
...
+
+## FAQ
+
+### I tried to use io_uring but got `java.lang.RuntimeException: failed to create io_uring ring fd Cannot allocate memory`
+
+When you tried to use io_uring but saw an exception that looked like this please try to encreate the [memlock limit](https://access.redhat.com/solutions/61334) for the user that tries to use io_uring:
+
+
+```
+Exception in thread "main" java.lang.UnsatisfiedLinkError: failed to load the required native library
```
Comment (chrisvest) [inline README.md:37]: Kind of odd from a users perspective that this becomes an UnsatisfiedLinkError.

Comment (normanmaurer) [reply README.md:37]: Well it happens during loading of the native lib, and its consistent with how we signal these failures back in other native libs in netty.

File: README.md (hunk 1 above)
Comment (normanmaurer) [reply README.md]: done

**Review by chrisvest — APPROVED** [review]

**Pull request merged**